runtime/security.go

```diff
@@ -197,8 +201,14 @@ func (p *securityEngine) resolveSecurity(instanceID, environment string, vars ma
 		return ResolvedSecurityOpen, nil
 	}
 
+	rules, err := p.expandRules(context.Background(), instanceID, claims)
```

Comment:
Probably need to propagate the source `ctx` instead of using background. Otherwise some of these calls might hang for cancelled requests or during graceful shutdown.